lib/xayn_card_view/card_view.dart

```diff
@@ -121,7 +121,7 @@ class CardViewState extends State<CardView> {
       final w = isVerticalScroll ? null : cardSize;
       final h = isVerticalScroll ? cardSize : null;
 
-      if (widget.itemCount > 0) {
+      if (widget.itemCount > 0 && _index >= 0) {
```

Comment:
just wondering, is there a case where `_index` is negative? 🤔

Reply:
Yes. In the `_onDragEnd` method the index is calculated and can be negative.

Reply:
if so, we can `clamp` it there better, wdyt?

Reply:
The animation looks better if we use:
```
if (_index + pageOffset < 0) return;
```

rather than:
```
_index = (_index + pageOffset).clamp(0, widget.itemCount - 1);
```

Reply:
I see. 
this happens when `clamp`ing it in `_onDragEnd`; [here](https://github.com/xaynetwork/xayn_card_view/blob/TB-3055_exception_fix/lib/xayn_card_view/card_view.dart#L263)?

Reply:
Exactly.

Reply:
There is a part in onDragEnd where pageOffset is calculated.

But, it cannot be -1 if the _index is at 0,
at the same time, it cannot be +1 if the _index is equal to widget.itemCount - 1.

So, I would fix it there, as this PR only solves the issue for indices below 0, but not for the other side, when they are greater than widget.itemCount - 1

Reply:
@frankpepermans Fixed.